### PR TITLE
Fix Nesting Transformation

### PR DESCRIPTION
**Pull request by TizianoDeMatteis**
Fix for nesting transformation:
- previously only source and sink node in the SDFG were considered and moved to the nested one. Now all the input/output node are considered
- change of transient property if an array has been moved to a Nested SDFG

### CODE REVIEW

File: dace/transformation/interstate/sdfg_nesting.py [hunk 1]
```diff
@@ -117,24 +94,25 @@ def apply(self, sdfg):
         for oldarrname, newarrname in transients.items():
             nested_sdfg.arrays.pop(oldarrname)
             nested_sdfg.arrays[newarrname].transient = False
-            outer_sdfg.arrays[oldarrname].transient = False
+            outer_sdfg.arrays[oldarrname].transient = True
```
Comment (tbennun) [inline dace/transformation/interstate/sdfg_nesting.py]: What's this doing? Do we even need this line, e.g. in the case that the outer array is not transient?

Comment (TizianoDeMatteis) [reply dace/transformation/interstate/sdfg_nesting.py]: Probably it is not needed. Putting it to `False` was wrong by the way, as additional kernel parameters were created

Comment (tbennun) [reply dace/transformation/interstate/sdfg_nesting.py]: ok. Can you try removing this line and see if the tests still pass?

Comment (TizianoDeMatteis) [reply dace/transformation/interstate/sdfg_nesting.py]: I confirm that that was unecessary

**Pull request merged**